pallets/identity/src/lib.rs

```diff
@@ -2082,6 +2086,67 @@ impl<T: Trait> Module<T> {
         }
     }
 
+    /// Use to get the filtered authorization data for a given signatory
+    /// - if auth_type is None then return authorizations data on the basis of the `allow_expired` boolean
+    /// - if auth_type is Some(value) then return filtered authorizations on the value basis type in conjunction
+    ///   with `allow_expired` boolean condition
+    pub fn get_filtered_authorizations(
+        signatory: Signatory<T::AccountId>,
+        allow_expired: bool,
+        auth_type: Option<AuthorizationType>,
+    ) -> Vec<Authorization<T::AccountId, T::Moment>> {
+        let now = <pallet_timestamp::Module<T>>::get();
+
+        if let Some(type_of_auth) = auth_type {
+            <Authorizations<T>>::iter_prefix_values(signatory)
+                .filter(|auth| {
+                    if !allow_expired {
+                        if let Some(expiry) = auth.expiry {
+                            if expiry < now {
+                                return false;
+                            }
+                        }
+                    }
+                    match auth.authorization_data {
+                        AuthorizationData::AttestMasterKeyRotation(..) => {
+                            type_of_auth == AuthorizationType::AttestMasterKeyRotation
+                        }
```

Comment:
Instead of matching it here, we can define a `get_type` function for `AuthorizationData` that does the matching and returns the type. We will then only need to do an equality check here.